src/main/fc/fc_msp.c

```diff
@@ -1866,7 +1867,11 @@ static bool processInCommand(uint8_t cmdMSP)
         masterConfig.gyro_sync_denom = read8();
         masterConfig.pid_process_denom = read8();
         masterConfig.motorConfig.useUnsyncedPwm = read8();
-        masterConfig.motorConfig.motorPwmProtocol = read8();
+#ifdef USE_DSHOT
+        masterConfig.motorConfig.motorPwmProtocol = constrain(read8(), 0, PWM_TYPE_DSHOT150);
+#else
+        masterConfig.motorConfig.motorPwmProtocol = constrain(read8(), 0, PWM_TYPE_BRUSHED);
+#endif
```

Comment:
This is fragile and ugly. Instead update `motorPwmProtocolTypes_e` as below
```
typedef enum {
    PWM_TYPE_STANDARD = 0,
    PWM_TYPE_ONESHOT125,
    PWM_TYPE_ONESHOT42,
    PWM_TYPE_MULTISHOT,
    PWM_TYPE_BRUSHED,
    PWM_TYPE_DSHOT600,
    PWM_TYPE_DSHOT150.
    PWM_TYPE_MAX = PWM_TYPE_DSHOT150
} motorPwmProtocolTypes_e;
```
and use
```
masterConfig.motorConfig.motorPwmProtocol = constrain(read8(), 0, PWM_TYPE_MAX);
```